test/util/GTestHelpers.h

```diff
@@ -124,3 +124,27 @@ MATCHER_P(InsertIntoStream, matcher,
   *result_listener << "that yields \"" << output << "\"";
   return testing::ExplainMatchResult(matcher, output, result_listener);
 }
+
+// Helper type that allows to use non-copyable types in gtest matchers.
+template <typename T>
+class CopyShield {
+  std::shared_ptr<T> pointer_;
+
+ public:
+  explicit CopyShield(T data)
+      : pointer_{std::make_shared<T>(std::move(data))} {}
+
+  template <typename... Ts>
+  explicit CopyShield(Ts&&... args)
+      : pointer_{std::make_shared<T>(AD_FWD(args)...)} {}
```

Comment:
This needs a `requires  that `T` is constructible from the `Ts&&...`, otherwise this hides
the copy and move constructors of `CopyShield`, which then leads to miscompilations.
(Sonarcloud normally explains, but probably omits the `test/...` files in its analysis.